src/core/lib/support/murmur_hash.cc

```diff
@@ -45,7 +45,8 @@ uint32_t gpr_murmur_hash3(const void *key, size_t len, uint32_t seed) {
 
   /* body */
   for (i = -(int)nblocks; i; i++) {
-    memcpy(&k1, blocks + i, sizeof(uint32_t));
+    memcpy(&k1, (const uint8_t*)(blocks) + i * sizeof(uint32_t),
```

Comment:
So we're still in UB land: the creation of the blocks variable itself is UB.

We'd want on line 43:
```
const uint8_t* blocks = ((const uint8_t*)key) + nblocks * sizeof(uint32_t);
```

Reply:
Also, I haven't properly thought through whether this is a strict aliasing violation or not.

Reply:
That's true, fixed. Not sure what the strict aliasing violation here would be however.